src/sentry/api/endpoints/organization_events.py

```diff
@@ -353,91 +364,180 @@ def _data_fn(scopedDataset, offset, limit, query) -> dict[str, Any]:
                 on_demand_metrics_type=on_demand_metrics_type,
             )
 
-        def data_fn_factory(scopedDataset):
-            """
-            This factory closes over query and dataset in order to make an additional request to the errors dataset
-            in the case that this request is from a dashboard widget and we're trying to split their discover dataset.
+        def _dashboards_data_fn(scopedDataset, offset, limit, scoped_query, dashboard_widget_id):
```

Comment:
Can we add the sentry_sdk.tracing decorator for this function so we get a span saying how long it took?